tests/func/test_gc.py

```diff
@@ -401,3 +401,22 @@ def test_gc_all_experiments(tmp_dir, scm, dvc):
     assert (
         tmp_dir / ".dvc" / "cache" / baz_hash[:2] / baz_hash[2:]
     ).read_text() == "baz"
+
+
+def test_gc_rev_num(tmp_dir, scm, dvc):
```

Comment:
As we had already had a unit test for `iter_revs` with `--num` and `--rev`, maybe we only need a mocked test?

Reply:
> maybe we only need a mocked test?

Maybe it would be enough, although given that the added one is not expensive I prefer to have a somehow redundant but functional test